### ISSUE TITLE
Model level validators don't work as expected

### ISSUE THREAD

**Issue by synic** [schematics/schematics#444]
If you have a Model, with a validate function like this:

``` python
class User(Model):
    email = types.StringType()
    def validate_email(self, data, value):
        return value.lower()
```

Returning the value does NOT affect the field on the object, or in `_data`.

So, for example, this would be False:

``` python
u = User({'email': 'TEST@TEST.com'})
u.email = 'test@test.com'
```

This, however, does work:

``` python
def validate_email(self, data, value):
    data['email'] = value.lower()
```

I've tried both 1.1.1 and the current pre-release.

**Comment by bintoro** [comment]
At least for now, this is by design; see #206 and #355.

Currently, one way to accomplish what you want to do is implement an `EmailType` that derives from `StringType` and does `value.lower()` in its `to_native()` method after calling `super().to_native()`.

Unless we reverse the decision not to have validators modify a value, I think I would like there to be a cleanup stage between the conversion and validation steps, so you could define a `sanitize_email()` method without implementing a custom type. #206 is still open and looks like the right place to track this issue.

**Comment by synic** [comment]
Ok, sounds good.  What is the status of the current pre-release version?  Any timeline or roadmap available?

**Comment by lkraider** [comment]
Closed as of #206.
